proposals/extensions.md

```diff
@@ -715,6 +956,30 @@ The search proceeds as follows:
 - If no extension indexer is found to be suitable for the element access
   in any enclosing scope, a compile-time error occurs.
 
+#### This access
+
+We modify the [this access rules](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#12813-this-access) as follows:
+
+A this_access has one of the following meanings:
+- [...]
+- ***When this is used in a primary_expression within an instance method or instance accessor of an extension with a class underlying type,
+  it is classified as a value. The type of the value is the instance type of the extension within which the usage occurs, 
+  and the value is a reference to the object for which the method or accessor was invoked.**
+- ***When this is used in a primary_expression within an instance method or instance accessor of an extension with a struct underlying type, 
```

Comment:
Need to specify the meaning for a scenario when the underlying type is a type parameter not known to be a reference type or a value type.

Reply:
Yes. I meant to have this as an open issue but forgot. Added.